xdem/workflows/workflows.py

```diff
@@ -0,0 +1,196 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Workflow class
+"""
+
+import csv
+import logging
+import os
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Dict
+
+import geoutils as gu
+import matplotlib.pyplot as plt
+import numpy as np
+import yaml  # type: ignore
+from geoutils import Mask
+from geoutils.raster import RasterType
+from yaml.dumper import SafeDumper  # type: ignore
+
+import xdem
+from xdem import DEM
+from xdem.coreg.base import InputCoregDict, OutputCoregDict
+from xdem.workflows.schemas import validate_configuration
+
+
+class Workflows(ABC):
+    """
+    Abstract Class for workflows
+    """
+
+    def __init__(self, user_config: str | Dict[str, Any]) -> None:
+        """
+        Initialize the workflows class
+        :param user_config: str path to a config file or dict as config
+        :return: None
+        """
+
+        # Load configuration
+        if isinstance(user_config, str):
+            if not os.path.isfile(user_config):
+                raise FileNotFoundError(f"{user_config} does not exist")
+            self.config_path = user_config
+            config_not_verify = self.load_config()
+        elif isinstance(user_config, dict):
+            config_not_verify = user_config
+        else:
+            raise ValueError(
+                "The configuration should be provided either as a path to the configuration file"
+                " or as a dictionary containing the configuration details."
+            )
+
+        self.config = validate_configuration(config_not_verify, self.schema)
+        self.level = self.config["outputs"]["level"]
+
+        self.outputs_folder = Path(self.config["outputs"]["path"])
+        self.outputs_folder.mkdir(parents=True, exist_ok=True)
+
+        for folder in ["png", "raster", "csv"]:
+            Path(self.outputs_folder / folder).mkdir(parents=True, exist_ok=True)
+
+        self.dico_to_show = [
+            ("Information about inputs", self.config["inputs"]),
+        ]
+
+    class NoAliasDumper(SafeDumper):  # type: ignore
+        """
+        NoAliasDumper to avoid id in YAML file
+        """
+
+        def ignore_aliases(self, data: Any) -> bool:
+            """
+            avoid id in YAML file
+            """
+            return True
+
+    def load_config(self) -> Dict[str, Any]:
+        """
+        Load a configuration file
+        :return: Configuration dictionary
+        """
+        if not os.path.exists(self.config_path):
+            raise FileNotFoundError(f"File not found : {self.config_path}")
+        with open(self.config_path) as f:
+            return yaml.safe_load(f)
+
+    def generate_graph(self, dem: RasterType, title: str, mask_path: str = None, **kwargs: Any) -> None:
```

Comment:
I think "graph" is used more often for a line plot showing a relationship between variables.
Maybe `generate_plot` or `generate_map`?